tests/test_experiment.py

```diff
@@ -0,0 +1,361 @@
+import pyspark.sql.functions as F
+import pytest
+
+from mozanalysis.experiment import Experiment
+from mozanalysis.utils import add_days
+
+
+def test_check_windows():
+    exp = Experiment('a-stub', '20190101')
+    exp_8d = Experiment('experiment-with-8-day-cohort', '20190101', 8)
+
+    # When we have complete data for 20190114...
+    the_fourteenth = '20190114'
+
+    # ...we have 14 dates of data for those who enrolled on the 1st
+    exp._print_enrollment_window(the_fourteenth, 14)
+
+    # We don't have 14 dates of data for the 8-day cohort:
+    with pytest.raises(AssertionError):
+        exp_8d._print_enrollment_window(the_fourteenth, 14)
```

Comment:
hm, these validations don't feel like something that ought to be the responsibility of something named `_print` -- are these tests all redundant with the next method? If these tests don't provide additional coverage, maybe they're safe to delete

Reply:
Those validations are the responsibility of `_get_scheduled_max_enrollment_date()`. The output of said method is used:
* In `get_enrollments()` to filter the enrollments dataframe before returning it,
* In `filter_enrollments_for_conv_window()` (called in `get_per_client_data()`) to filter the enrollments dataframe before joining to `df`
* In `_print_enrollment_window()` to inform the user which date is going to be used by `filter_enrollments_for_conv_window()`.

I do think that the validation should happen at the time that this date is calculated, so this validation does belong in `_get_scheduled_max_enrollment_date()`.

w.r.t. the automated test suite, sure, this set of tests doesn't capture much of value that's not captured by `test_get_last_enrollment_date()`, so I'll delete the vast majority of the tests in `test_check_windows()` and rename it.